tests/test_core.py

```diff
@@ -68,6 +68,22 @@ def test_ExpressionTuple(capsys):
         ExpressionTuple((print, "hi")).eval_obj
 
 
+def test_eval_apply_fn():
+    class Sub(object):
+        def __call__(self, *args):
+            return args
+
+        def apply(self, *args):
```

Comment:
This test should probably make the signatures between the two methods distinct and confirm that the correct one is used.

Reply:
Good point

Reply:
I updated the test and added a docstring `_eval_apply_fn`